copy/multiple.go

```diff
@@ -140,7 +151,49 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 				UpdateDigest:    updatedManifestDigest,
 				UpdateSize:      int64(len(updatedManifest)),
 				UpdateMediaType: updatedManifestType})
-		default:
+		}
+		if instance.op == instanceCopyClone {
+			processedOp = false
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			unparsedInstance := image.UnparsedInstance(c.rawSource, &instanceCopyList[i].sourceDigest)
+			src, err := image.FromUnparsedImage(ctx, options.SourceCtx, unparsedInstance)
+			if err != nil {
+				return nil, fmt.Errorf("initializing image from source %s: %w", transports.ImageName(c.rawSource.Reference()), err)
+			}
+			imgConfig, err := src.OCIConfig(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("parsing image configuration: %w", err)
+			}
+			imgPlatform := imgspecv1.Platform{Architecture: imgConfig.Architecture,
+				OS:         imgConfig.OS,
+				OSVersion:  imgConfig.OSVersion,
+				OSFeatures: imgConfig.OSFeatures,
+				Variant:    imgConfig.Variant}
+			addAnnotations := make(map[string]string)
+			if instance.forceZstd {
+				// add `zstd` annotation
+				addAnnotations[internalManifest.OCI1InstanceAnnotationCompressionZSTD] = "true"
+				// copy options with new compression format
+				optionsClone := options
+				optionsClone.DestinationCtx.CompressionFormat = &compression.Zstd
```

Comment:
This doesn’t actually work well enough right now, because we reuse blobs at the destination regardless of compression. So copying a gzip image followed by trying to create a zstd clone would just create another image that links to the same just-uploaded gzip blobs.

We will need the blob reuse code around `copyLayer` to accept an option to use _only_ the requested compression format, and not any other. (I guess we would still ignore the compression level request if a blob exists with the same algorithm but a different level, just because we don’t have a cheap way to determine the compression level.)